### PR TITLE
Expand CP.61 to talk about the general "factory" pattern.

### PR DESCRIPTION
**Pull request by Quuxplusone**
This PR is along the same lines as #1610. The general rule is "don't touch raw resources with your hands; use an RAII factory instead."

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -14974,25 +14975,63 @@ There is no explicit locking and both correct (value) return and error (exceptio
...
+        std::cout << f1.get() + f2.get() << '\n';
+    }
+
+##### Example (good)
+
+This example shows one way you could follow the general pattern set by
+`std::async`, in a context where `std::async` itself was unacceptable for
+use in production.
+
+    void async_example(WorkQueue *wq)
```
Comment (cubbimew) [inline CppCoreGuidelines.md:15021]: why not WorkQueue&?

Comment (Quuxplusone) [reply CppCoreGuidelines.md:15021]: Just habit. In general, pass-by-reference is an invisible optimization of (the default) pass-by-value; you should use pass-by-(const)-reference only when it's _not important_ for the reader of the call-site to understand whether you're passing by value or by reference.
(This habit "avoid pass-by-mutable-reference" is in keeping with common industry practice, but not with the Guidelines AFAIK.)

Comment (cubbimew) [reply CppCoreGuidelines.md:15021]: different industries - different practice.
But to stay consistent within the document, F.17 wants this to be a reference, F.23 enforcement would flag the wq->, and R.2 note again points references as desirable.

Comment (Quuxplusone) [reply CppCoreGuidelines.md]: Okay, updated.

**Comment by hsutter** [comment]
Editors call: Thanks!

**Pull request merged**